Realm/Realm/Linq/RealmResultsVisitor2.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Newtonsoft.Json;
+
+namespace Realms
+{
+    internal class RealmResultsVisitor2 : ExpressionVisitor
+    {
+        private readonly Realm _realm;
+        private readonly RealmObjectBase.Metadata _metadata;
+        private QueryModel _query;
+
+        private QueryHandle _coreQueryHandle;
+        private SortDescriptorHandle _sortDescriptor;
```

Comment:
I think that we are not using this

Reply:
I'll keep those for compiling purposes since MakeResultsForQuery method uses these. I'll remove them when the CreateResults method is rewritten

Reply:
Now it's probably the right time to remove them :)